### PR TITLE
Dotty 0.4.0-RC1 Blog Post

### PR DESCRIPTION
**Pull request by allanrenucci**

### CODE REVIEW

**Comment by allanrenucci** [comment]
@felixmulder Do you want to write something about the new REPL?
@smarter Do you want to write something about Scala 2.12 support?

**Comment by felixmulder** [comment]
@allanrenucci - sure. Also, I've been thinking of some improvements:

- If a state is thrown away, and a new state used in its stead. That should be enough to reset the REPL. So, some mutability in `ReplDriver` init as well as `Rendering` should be moved into the session CC.

A difference between the two implementations that we've fixed:

```scala
scala> implicit val x: String = "wrong"
scala> implicit val y: String = "right"
scala> implicitly[String]
val res0: String = "right"
```

So, I guess that should be mentioned. Otherwise I'm not sure what more to add :)

**Comment by allanrenucci** [comment]
@smarter @felixmulder Any update on this?

**Comment by allanrenucci** [comment]
>So, I guess that should be mentioned. Otherwise I'm not sure what more to add :)

Autocomplete?

**Comment by smarter** [comment]
Still on my TODO list to write something, but I guess it's not too urgent if we're delaying the release a week :).

**Comment by smarter** [comment]
I just added a bunch of stuff, please proofread/improve as you see fit :).

**Comment by allanrenucci** [comment]
@heathermiller Could you please have a look? I'll open a PR to scala-lang if we are happy with this version.

File: docs/blog/_posts/2017-10-02-fourth-dotty-milestone-release.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+### Performance work
+Over the last few weeks, we started working on compilation speed with some good results:
+compiling [ScalaPB](https://github.com/dotty-staging/scalapb) is now 20% faster
+than with Dotty 0.3.0-RC2. You can follow along our progress on
+http://dotty-bench.epfl.ch/.
+
+
+## Trying out Dotty
+### Scastie
```
Comment (heathermiller) [inline docs/blog/_posts/2017-10-02-fourth-dotty-milestone-release.md]: Maybe you should link to an already-configured Dotty snippet? Else the default is Scala 2.12.x...

Comment (heathermiller) [reply docs/blog/_posts/2017-10-02-fourth-dotty-milestone-release.md]: Oh, oops, you already do that. I should've looked at the URL for this link. 🙈

File: docs/blog/_posts/2017-10-02-fourth-dotty-milestone-release.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+[Gitter channel](https://gitter.im/lampepfl/dotty). If you encounter a bug, please
+[open an issue on GitHub](https://github.com/lampepfl/dotty/issues/new).
+
+## Contributing
+Thank you to all the contributors who made this release possible!
+
+According to `git shortlog -sn --no-merges 0.3.0-RC1..0.4.0-RC1` these are:
+
+```
+   TODO
```
Comment (heathermiller) [inline docs/blog/_posts/2017-10-02-fourth-dotty-milestone-release.md]: I presume the PR to scala-lang will include this? ;-)

File: docs/blog/_posts/2017-10-02-fourth-dotty-milestone-release.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+```
+
+If you want to get your hands dirty with any of this, now is a good moment to get involved!
```
Comment (heathermiller) [inline docs/blog/_posts/2017-10-02-fourth-dotty-milestone-release.md]: > If you want to get your hands dirty with any of this, now is a good moment to get involved!

I would instead say:

> If you want to get your hands dirty and contribute to Dotty, now is a good time to get involved!

File: docs/blog/_posts/2017-10-02-fourth-dotty-milestone-release.md [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+You can have a look at our [Getting Started page](http://dotty.epfl.ch/docs/contributing/getting-started.html),
```
Comment (heathermiller) [inline docs/blog/_posts/2017-10-02-fourth-dotty-milestone-release.md]: > You can have a look at our [Getting Started page](http://dotty.epfl.ch/docs/contributing/getting-started.html)

I would say:

> You can have a look at our [Getting Started page for new contributors](http://dotty.epfl.ch/docs/contributing/getting-started.html)

File: docs/blog/_posts/2017-10-02-fourth-dotty-milestone-release.md [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+our [Awesome Error Messages](http://scala-lang.org/blog/2016/10/14/dotty-errors.html) or some of
```
Comment (heathermiller) [inline docs/blog/_posts/2017-10-02-fourth-dotty-milestone-release.md]: > our [Awesome Error Messages](http://scala-lang.org/blog/2016/10/14/dotty-errors.html)

I would say:

> the [Awesome Error Messages](http://scala-lang.org/blog/2016/10/14/dotty-errors.html) project

File: docs/blog/_posts/2017-10-02-fourth-dotty-milestone-release.md [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+the simple [Dotty issues](https://github.com/lampepfl/dotty/issues?q=is%3Aissue+is%3Aopen+label%3Aexp%3Anovice).
+They make perfect entry-points into hacking on the compiler.
+
+We are looking forward to having you join the team of contributors.
+
+## Library authors: Join our community build
+Dotty now has a set of libraries that are built against every nightly snapshot.
```
Comment (heathermiller) [inline docs/blog/_posts/2017-10-02-fourth-dotty-milestone-release.md]: > Dotty now has a set of libraries that are built against every nightly snapshot.

I would say:

> Dotty now has a set of widely-used community libraries that are built against every nightly Dotty snapshot.

File: docs/blog/_posts/2017-10-02-fourth-dotty-milestone-release.md (hunk 2 above)
Comment (allanrenucci) [reply docs/blog/_posts/2017-10-02-fourth-dotty-milestone-release.md]: Waiting for a couple more PRs to get in before I tag the release. Will definitely be in the scala-lang PR =)

**Comment by allanrenucci** [comment]
Thanks @heathermiller for your review!

**Pull request merged**